tests/PhpLeagueAccessTokenProviderTest.php

```diff
@@ -189,6 +192,28 @@ public function testGetAuthTokenWithInsecureUrlDoesntReturnAccessToken(): void
         $this->assertNull($this->defaultTokenProvider->getAuthorizationTokenAsync('http://example.com')->wait());
     }
 
+    public function testGetAccessTokenWithLocalhostStringWithHttpReturnsAccessToken(): void
+    {
+        $tokenRequestContext  = new ClientCredentialContext('tenant', 'client', 'secret');
+        foreach(PhpLeagueAccessTokenProvider::LOCALHOST_STRINGS as $host) {
+            $mockResponses = [
+                new Response(200, [], json_encode(['access_token' => $this->testJWT, 'expires_in' => 0.1])),
+                function (Request $request) use ($tokenRequestContext, $host) {
+                    parse_str($request->getBody()->getContents(), $requestBodyMap);
+                    $expectedBody = array_merge($tokenRequestContext->getParams(), [
+                        'scope' => "http://$host/.default"
+                    ]);
+                    $this->assertEquals($expectedBody, $requestBodyMap);
+                    return new Response(200, [], json_encode(['access_token' => 'xyz', 'expires_in' => 1]));
+                },
+            ];
+            $tokenProvider = new PhpLeagueAccessTokenProvider($tokenRequestContext, ["https://$host/.default"]);
+            $tokenProvider->getOauthProvider()->setHttpClient($this->getMockHttpClient($mockResponses));
+            $token         = $tokenProvider->getAuthorizationTokenAsync("http://$host")->wait();
```

Comment:
Need a second token request to trigger the request body validation callback
```suggestion
            $token         = $tokenProvider->getAuthorizationTokenAsync("http://$host")->wait();
            $token         = $tokenProvider->getAuthorizationTokenAsync("http://$host")->wait();
```